ballista/rust/core/src/serde/physical_plan/from_proto.rs

```diff
@@ -121,53 +120,64 @@ impl TryInto<Arc<dyn ExecutionPlan>> for &protobuf::PhysicalPlanNode {
             }
             PhysicalPlanType::CsvScan(scan) => {
                 let schema = Arc::new(convert_required!(scan.schema)?);
-                let options = CsvReadOptions::new()
-                    .has_header(scan.has_header)
-                    .file_extension(&scan.file_extension)
-                    .delimiter(scan.delimiter.as_bytes()[0])
-                    .schema(&schema);
                 let projection = scan.projection.iter().map(|i| *i as usize).collect();
-                Ok(Arc::new(CsvExec::try_new(
-                    &scan.path,
-                    options,
+                let statistics = convert_required!(scan.statistics)?;
+
+                Ok(Arc::new(CsvExec::new(
+                    Arc::new(LocalFileSystem {}),
+                    scan.files
+                        .iter()
+                        .map(|f| f.try_into())
+                        .collect::<Result<Vec<PartitionedFile>, _>>()?,
+                    statistics,
+                    schema,
+                    scan.has_header,
+                    str_to_byte(&scan.delimiter)?,
                     Some(projection),
                     scan.batch_size as usize,
-                    None,
-                )?))
+                    scan.limit.as_ref().map(|sl| sl.limit as usize),
```

Comment:
My goal here was to serialize the whole object, I wasn't aiming any specific bugfix. I don't think this was a bug earlier in the sense that the final result would be correct because the limit is also ensured by the `LimitExec`. But it was surprising indeed that the `CsvExec` was different before and after serialization 😄